src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -1834,9 +1866,23 @@ private DatabaseItemType GetType(string type)
             return DatabaseItemType.Unknown;
         }
 
+        public Stream GetAttachmentStream(LazyStringValue hash, out string tag)
+        {
+            tag = null;
+            return null;
+        }
+
+        public Task<DatabaseRecord> GetShardedDatabaseRecordAsync()
+        {
+            // Used only in Database Source
+            throw new NotSupportedException("GetShardedDatabaseRecordAsync is not supported in Stream Source, " +
+                                            "it is only supported from Sharded Database Source.");
+        }
+
         public void Dispose()
         {
             _peepingTomStream.Dispose();
+            _byteStringContext.Dispose();
```

Comment:
You aren't handling the `_toDispose` here?

Reply:
No. I need to handle it after we write to destination